feat: parenthetical syntax for message sends (`cycles`, `timeout` etc.) (#4608)
dfinity/motoko: doc/md/writing-motoko/async-data.md

Awaiting a future a second time will just produce the same result, including re-throwing any error stored in the future. Suspension occurs even if the future is already complete; this ensures state changes and message sends prior to every `await` are committed.

## Using parentheticals to modify message send modalities

In the above examples all messages sent to the `Counter` actor are of unbounded-wait kind and attach no cycles. Both of these

> aspects can be changed by syntactically adding a parenthetical and thus modifying the modality of the send.

Reviewer: ```suggestion
aspects can be configured by syntactically adding a parenthetical and thus modifying the dynamic attributes of the message send.
```

To add cycles to the send one would write
``` motoko no-repl
let a = (with cycles = 42_000_000) Counter.bump();
```
Similarly, one can specify a cutoff for potentially long-running messaging by providing a timeout, and thus delivering a bounded-wait

> message to the destination actor

Reviewer: Similarly, one can specify an explicit timeout to apply when awaiting the result of the message. This is useful in applications that prefer best-effort  over guaranteed response message delivery:

> Similarly, one can specify a cutoff for potentially long-running messaging by providing a timeout, and thus delivering a bounded-wait
> message to the destination actor

Reviewer: ```suggestion
Similarly, one can specify a cutoff for potentially long-running messaging by providing a timeout, and thus delivering a bounded-wait
message to the destination actor
```

> Similarly, one can specify a cutoff for potentially long-running messaging by providing a timeout, and thus delivering a bounded-wait
> message to the destination actor

Reviewer: ```suggestion
Similarly, one can specify an explicit timeout to apply when awaiting the result of the message. This is useful in applications that prefer best-effort  over guaranteed response message delivery:
```

Author: I have committed this already.

> Similarly, one can specify a cutoff for potentially long-running messaging by providing a timeout, and thus delivering a bounded-wait
> message to the destination actor

Author: ```suggestion
Similarly, one can specify an explicit timeout to apply when awaiting the result of the message. This is useful in applications that prefer best-effort over guaranteed response message delivery:
```
